src/main/java/uk/gov/pay/webhooks/webhook/resource/WebhookResource.java

```diff
@@ -103,7 +107,39 @@ public SigningKeyResponse regenerateSigningKey(@PathParam("externalId") @NotNull
                     .map(WebhookResponse::from)
                     .toList();
     }
-    
+
+    @UnitOfWork
+    @Path("/{externalId}/messages")
+    @GET
+    public WebhookMessageSearchResponse getWebhookMessages(
+            @PathParam("externalId") String externalId,
+            @QueryParam("page") Integer page,
+            @QueryParam("status") String status
```

Comment:
Can we fail the request if the `status` does not appear in the `DeliveryStatus` enum?

Reply:
Yeah doing that explicitly sounds like a good idea. 

At the moment it will just return an empty set as nothing will have that status, i.e 
```
{ 
  count: 0,
  page: 1,
  total: 0,
  results: []
}
```